File: problem21.cpp
#include <iostream>
using namespace std;
int calc(int x)
{
	int sum = 0;
	for (int i = 1; i <= (x >> 1); ++ i)
		if (x % i == 0) sum += i;
	return sum;
}
const int n = 10000;
int f[n];
int main()
{
	for (int i = 1; i <= n; ++ i)
		f[i] = calc(i);
	int ans = 0;
	for (int i = 1; i <= n; ++ i)
		if (f[i] <= n && f[i] > 0 && f[f[i]] == i && f[i] != i) ans += i;
	cout << ans << endl;
	return 0;
}

File: problem1-50/problem21/problem21.pas
program problem21;
const maxn=10000;
var f:array[0..maxn] of longint;
    i,ans:longint;
function calc(x:longint):longint;
 var i,s:longint;
 begin
     s:=0;
     for i:=1 to x-1 do
	if (x mod i=0) then s:=s+i;
     calc:=s;
 end;

begin
    f[0]:=0;
    for i:=1 to maxn do f[i]:=calc(i);
    ans:=0;
    for i:=1 to maxn do
	if (f[i]<=maxn) and (f[f[i]]=i) and (i<>f[i]) then ans:=ans+i;
    writeln(ans);
end.
